### ISSUE TITLE
Use pretrained instruct pix2pix in training of instruct pix2pix

### ISSUE THREAD

**Issue by mahdip72** [huggingface/diffusers#3455]
Can I use "timbrooks/instruct-pix2pix" model as the pre-trained weight in [train_instruct_pix2pix.py](https://github.com/huggingface/diffusers/blob/main/examples/instruct_pix2pix/train_instruct_pix2pix.py) code?

**Comment by sayakpaul** [comment]
Yes, we can. We will soon release a tutorial about it. Stay tuned :)

**Comment by mahdip72** [comment]
Great. I tried to train using the "timbrooks/instruct-pix2pix" weights. During inference I used this line of code:
`
new_image = pipeline(prompt=prompt, image=input_image,
                     num_inference_steps=50,
                     generator=generator).images[0]
`
However, despite the fact that my loss function goes down, the output of my model looks like this:
![1](https://github.com/huggingface/diffusers/assets/42680708/05cdf1df-1012-45af-a6c1-45a28cb6799a)

Did you encounter with this issue during training?

I used these settings:
mixed_precision: fp32
lora: disable
use_ema: False
train_batch_size: 8
gradient_accumulation_steps: 1
resolution: 256

Also, I load sub models using these lines:
`
tokenizer = CLIPTokenizer.from_pretrained(model_path, subfolder="tokenizer", revision=revision)
text_encoder = CLIPTextModel.from_pretrained(model_path, subfolder="text_encoder", revision=revision)
noise_scheduler = DDPMScheduler.from_pretrained(model_path, subfolder="scheduler")
vae = AutoencoderKL.from_pretrained(model_path, subfolder="vae", revision=revision)
unet = UNet2DConditionModel.from_pretrained(model_path, subfolder="unet", revision=non_ema_revision)
`

**Comment by sayakpaul** [comment]
Thanks for sharing!

Could you share the following details?

* Which dataset are you using?
* Which training script are you using? Because the current InstructPix2Pix training script doesn't support fine-tuning from `"timbrooks/instruct-pix2pix"`. 
* Have you taken a look at the discussion on the expected results [here](https://huggingface.co/docs/diffusers/training/instructpix2pix)? 

Also, the loss values are kind of uninformative for diffusion-related training. It's better to have qualitative validation in between your training epochs to see if the model is learning something. But for improving the reliance of loss to gauge performance, you could try using Min-SNR. We support training with it [here](https://huggingface.co/docs/diffusers/training/text2image#training-with-minsnr-weighting). I believe it should be possible to extend that easily to our InstructPix2Pix training script :) 

> https://github.com/huggingface/diffusers/issues/3455#issuecomment-1551102568

Regarding the ^ comment, feel free to check out https://github.com/huggingface/blog/pull/1125. It has many piece of interesting discussions that you may find helpful for your use case :)

**Comment by sayakpaul** [comment]
> https://github.com/huggingface/diffusers/issues/3455#issue-1712702443


@mahdip72, here you: https://huggingface.co/blog/instruction-tuning-sd (w.r.t my comments in https://github.com/huggingface/diffusers/issues/3455#issuecomment-1551102568)

**Comment by mahdip72** [comment]
@sayakpaul  I use a custom dataset with my custom training code. I switched to the diffuser code and my problem fixed. I change the code to load "timbrooks/instruct-pix2pix" weights and it works okay. However, the original implementation used EulerAncestralDiscrete Scheduler with 100 steps. I tested it and my training was not stable with that scheduler. So, I used default DDPM Scheduler. 
Thanks again for your support! I will check the instruction tuning sd training code.

**Comment by sayakpaul** [comment]
Closing the issue then. Feel free to share your results on our forum:
https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63

**Comment by irshadtech10** [comment]
> ###
> [May 17, 2023](#issue-1712702443)

I am looking for the answer as well
